### ISSUE TITLE
TabView.CloseButtonOverlayMode not set on new tabs

### ISSUE THREAD

**Issue by Mangepange** [microsoft/microsoft-ui-xaml#9180]
### Describe the bug

The value of TabView.CloseButtonOverlayMode is not reflected on new tabs added in runtime. New tabs always have their close button visible.

### Steps to reproduce the bug

Using the template studio for WinUI, i updated MainPage.xaml and MainViewModel with the code below.
Tabs added with the "Add"-button show their close buttons.
Switching CloseButtonOverlayMode in runtime (which btw does not seem to respond to binding) seems to effect the existing tabs, but when adding more tabs they always show the close button.

MainViewModel:

    using System.Collections.ObjectModel;
    using CommunityToolkit.Mvvm.ComponentModel;
    using CommunityToolkit.Mvvm.Input;

    namespace App2.ViewModels;
    public partial class MainViewModel : ObservableRecipient
    {
        [ObservableProperty]
        private ObservableCollection<string> _tabs = new();

        private int _counter = 0;

        [RelayCommand]
        private void Add()
        {
            Tabs.Add($"Content {_counter++}");
        }
    }

MainPage.xaml:

    <Page
        x:Class="App2.Views.MainPage"
        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d">
        <StackPanel>
            <Button Command="{x:Bind ViewModel.AddCommand}" Content="Add"/>
            <TabView TabItemsSource="{x:Bind ViewModel.Tabs}"
                     CloseButtonOverlayMode="OnPointerOver"/>
        </StackPanel>
    </Page>

### Expected behavior

Added TabItems in a TabView should respect the CloseButtonOverlayMode property.

### Screenshots

![image](https://github.com/microsoft/microsoft-ui-xaml/assets/10757343/82b3e5ff-b9b4-4c28-a62c-2811e958d066)

### NuGet package version

WinUI 3 - Windows App SDK 1.4.3: 1.4.231115000

**Comment by twinkles-twinstar** [comment]
Observed the same issue on WinAppSDK 1.5.
I need to assign a value to `CloseButtonOverlayMode` every time I add an item to TabView.
It seems that this problem has existed for at least two years, but MS has not fixed it.

**Comment by Mangepange** [comment]
Yeah, I fixed it with a Behavior that sets CloseButtonOverlayMode every time TabItemsChanged is triggered.
